File: chapter7/KotlinOOP/src/test/java/com/gavinandre/kotlinoop/DelegationDemoTest.java
package com.gavinandre.kotlinoop;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

@RunWith(JUnit4.class)
public class DelegationDemoTest {

    @Test
    public void test() {
        JRealSubject jRealSubject = new JRealSubject();
        JProxy jProxy = new JProxy(jRealSubject);
        jProxy.request();
    }
}

File: chapter7/KotlinOOP/src/test/kotlin/com/gavinandre/kotlinoop/DelegationDemoTest.kt
package com.gavinandre.kotlinoop

import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4

@RunWith(JUnit4::class)
class DelegationDemoTest {

    @Test
    fun test() {
        val subject = RealSubject("World")
        subject.hello()
        println("----------------------")
        val proxySubject = ProxySubject(subject)
        proxySubject.hello()
    }
}